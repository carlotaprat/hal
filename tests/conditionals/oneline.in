def foo:
    if True: print x;
    elif y: print l
    else: print y

if True:
    pass;

print x if True else print y
print x, y, mult 3, 4 if foobar else print 2
