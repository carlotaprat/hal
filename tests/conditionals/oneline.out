digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FUNDEF"];
  n1 [label="FUNDEF"];
  n2 [label="foo"];
  n3 [label="PARAMS"];
  n4 [label="BLOCK"];
  n4 [label="BLOCK"];
  n5 [label="IF_STMT"];
  n5 [label="IF_STMT"];
  n6 [label="FUNCALL"];
  n6 [label="FUNCALL"];
  n7 [label="True"];
  n8 [label="ARGS"];
  n9 [label="BLOCK"];
  n9 [label="BLOCK"];
  n10 [label="FUNCALL"];
  n10 [label="FUNCALL"];
  n11 [label="print"];
  n12 [label="ARGS"];
  n12 [label="ARGS"];
  n13 [label="FUNCALL"];
  n13 [label="FUNCALL"];
  n14 [label="x"];
  n15 [label="ARGS"];
  n16 [label="BLOCK"];
  n16 [label="BLOCK"];
  n17 [label="IF_STMT"];
  n17 [label="IF_STMT"];
  n18 [label="FUNCALL"];
  n18 [label="FUNCALL"];
  n19 [label="y"];
  n20 [label="ARGS"];
  n21 [label="BLOCK"];
  n21 [label="BLOCK"];
  n22 [label="FUNCALL"];
  n22 [label="FUNCALL"];
  n23 [label="print"];
  n24 [label="ARGS"];
  n24 [label="ARGS"];
  n25 [label="FUNCALL"];
  n25 [label="FUNCALL"];
  n26 [label="l"];
  n27 [label="ARGS"];
  n28 [label="BLOCK"];
  n28 [label="BLOCK"];
  n29 [label="FUNCALL"];
  n29 [label="FUNCALL"];
  n30 [label="print"];
  n31 [label="ARGS"];
  n31 [label="ARGS"];
  n32 [label="FUNCALL"];
  n32 [label="FUNCALL"];
  n33 [label="y"];
  n34 [label="ARGS"];
  n35 [label="IF_STMT"];
  n35 [label="IF_STMT"];
  n36 [label="FUNCALL"];
  n36 [label="FUNCALL"];
  n37 [label="True"];
  n38 [label="ARGS"];
  n39 [label="BLOCK"];
  n39 [label="BLOCK"];
  n40 [label="FUNCALL"];
  n40 [label="FUNCALL"];
  n41 [label="pass"];
  n42 [label="ARGS"];
  n43 [label="IF_STMT"];
  n43 [label="IF_STMT"];
  n44 [label="FUNCALL"];
  n44 [label="FUNCALL"];
  n45 [label="foobar"];
  n46 [label="ARGS"];
  n47 [label="BLOCK"];
  n47 [label="BLOCK"];
  n48 [label="FUNCALL"];
  n48 [label="FUNCALL"];
  n49 [label="print"];
  n50 [label="ARGS"];
  n50 [label="ARGS"];
  n51 [label="FUNCALL"];
  n51 [label="FUNCALL"];
  n52 [label="x"];
  n53 [label="ARGS"];
  n54 [label="FUNCALL"];
  n54 [label="FUNCALL"];
  n55 [label="y"];
  n56 [label="ARGS"];
  n57 [label="FUNCALL"];
  n57 [label="FUNCALL"];
  n58 [label="mult"];
  n59 [label="ARGS"];
  n59 [label="ARGS"];
  n60 [label="3"];
  n61 [label="4"];
  n62 [label="BLOCK"];
  n62 [label="BLOCK"];
  n63 [label="FUNCALL"];
  n63 [label="FUNCALL"];
  n64 [label="print"];
  n65 [label="ARGS"];
  n65 [label="ARGS"];
  n66 [label="2"];

  n0 -> n1 // "BLOCK" -> "FUNDEF"
  n1 -> n2 // "FUNDEF" -> "foo"
  n1 -> n3 // "FUNDEF" -> "PARAMS"
  n1 -> n4 // "FUNDEF" -> "BLOCK"
  n4 -> n5 // "BLOCK" -> "IF_STMT"
  n5 -> n6 // "IF_STMT" -> "FUNCALL"
  n6 -> n7 // "FUNCALL" -> "True"
  n6 -> n8 // "FUNCALL" -> "ARGS"
  n5 -> n9 // "IF_STMT" -> "BLOCK"
  n9 -> n10 // "BLOCK" -> "FUNCALL"
  n10 -> n11 // "FUNCALL" -> "print"
  n10 -> n12 // "FUNCALL" -> "ARGS"
  n12 -> n13 // "ARGS" -> "FUNCALL"
  n13 -> n14 // "FUNCALL" -> "x"
  n13 -> n15 // "FUNCALL" -> "ARGS"
  n5 -> n16 // "IF_STMT" -> "BLOCK"
  n16 -> n17 // "BLOCK" -> "IF_STMT"
  n17 -> n18 // "IF_STMT" -> "FUNCALL"
  n18 -> n19 // "FUNCALL" -> "y"
  n18 -> n20 // "FUNCALL" -> "ARGS"
  n17 -> n21 // "IF_STMT" -> "BLOCK"
  n21 -> n22 // "BLOCK" -> "FUNCALL"
  n22 -> n23 // "FUNCALL" -> "print"
  n22 -> n24 // "FUNCALL" -> "ARGS"
  n24 -> n25 // "ARGS" -> "FUNCALL"
  n25 -> n26 // "FUNCALL" -> "l"
  n25 -> n27 // "FUNCALL" -> "ARGS"
  n17 -> n28 // "IF_STMT" -> "BLOCK"
  n28 -> n29 // "BLOCK" -> "FUNCALL"
  n29 -> n30 // "FUNCALL" -> "print"
  n29 -> n31 // "FUNCALL" -> "ARGS"
  n31 -> n32 // "ARGS" -> "FUNCALL"
  n32 -> n33 // "FUNCALL" -> "y"
  n32 -> n34 // "FUNCALL" -> "ARGS"
  n0 -> n35 // "BLOCK" -> "IF_STMT"
  n35 -> n36 // "IF_STMT" -> "FUNCALL"
  n36 -> n37 // "FUNCALL" -> "True"
  n36 -> n38 // "FUNCALL" -> "ARGS"
  n35 -> n39 // "IF_STMT" -> "BLOCK"
  n39 -> n40 // "BLOCK" -> "FUNCALL"
  n40 -> n41 // "FUNCALL" -> "pass"
  n40 -> n42 // "FUNCALL" -> "ARGS"
  n0 -> n43 // "BLOCK" -> "IF_STMT"
  n43 -> n44 // "IF_STMT" -> "FUNCALL"
  n44 -> n45 // "FUNCALL" -> "foobar"
  n44 -> n46 // "FUNCALL" -> "ARGS"
  n43 -> n47 // "IF_STMT" -> "BLOCK"
  n47 -> n48 // "BLOCK" -> "FUNCALL"
  n48 -> n49 // "FUNCALL" -> "print"
  n48 -> n50 // "FUNCALL" -> "ARGS"
  n50 -> n51 // "ARGS" -> "FUNCALL"
  n51 -> n52 // "FUNCALL" -> "x"
  n51 -> n53 // "FUNCALL" -> "ARGS"
  n50 -> n54 // "ARGS" -> "FUNCALL"
  n54 -> n55 // "FUNCALL" -> "y"
  n54 -> n56 // "FUNCALL" -> "ARGS"
  n50 -> n57 // "ARGS" -> "FUNCALL"
  n57 -> n58 // "FUNCALL" -> "mult"
  n57 -> n59 // "FUNCALL" -> "ARGS"
  n59 -> n60 // "ARGS" -> "3"
  n59 -> n61 // "ARGS" -> "4"
  n43 -> n62 // "IF_STMT" -> "BLOCK"
  n62 -> n63 // "BLOCK" -> "FUNCALL"
  n63 -> n64 // "FUNCALL" -> "print"
  n63 -> n65 // "FUNCALL" -> "ARGS"
  n65 -> n66 // "ARGS" -> "2"

}

